Avatar component improvements

fritzSF opened the pull request:
### Purpose
Enhance the Avatar component with configurable name display modes to support different customer requirements and improve UX flexibility.

### What has changed
- Added nameDisplay prop with three modes: tooltip (default), title (HTML title attribute), and inline (chip-style)
- Added tooltipFlow prop to control tooltip position ('top', 'left', 'right', 'down')
- Added textPosition prop for inline mode ('start', 'end')
- Fixed pointer-events issues on tooltips, status dots, and icons to prevent flickering
- Updated AvatarGroup to automatically convert inline mode to tooltip mode
- Added comprehensive documentation with examples and props reference
- Version bump to 5.2.4

### Steps to test
1. Navigate to Avatar component page in the examples
2. Test tooltip mode - hover to see tooltips with different positions (top, left, right, down)
3. Test inline mode - verify text appears beside avatar with proper positioning (start/end)
4. Test title mode - verify native browser tooltip appears on hover
5. Test AvatarGroup - verify groups work correctly and inline mode converts to tooltip
6. Verify no tooltip flickering when hovering over avatars with status indicators

### Screenshots
<img width="891" height="962" alt="Avatar_dsiplay_modes" src="https://github.com/user-attachments/assets/574c3a2a-12f3-4441-bfee-6952fd55a5ec" />

[STT-1508]


[STT-1508]: https://sofab.atlassian.net/browse/STT-1508?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

fritzSF committed:
[STT-1508] Avatar component improvements

fritzSF committed:
5.2.4

thecalcc commented on app/styles/_tooltips.scss:
```diff
@@ -422,4 +422,10 @@
\ No newline at end of file
+}
+
+.sd-popup-positioner {
```
what is this for?

fritzSF replied:
There is an issue with tooltips: if you manage to hover over them, they don't disappear. This fixes the issue and is applied only to tooltips.

thecalcc commented on app/styles/_avatar.scss:
```diff
@@ -371,18 +382,141 @@
+        gap: var(--gap-0-5);
+        .sd-avatar--inline__text {
+            --name-font-size: var(--text-size-xx-small);
+            padding-inline-end: 1em;
```
shouldn't we be using design tokens here for the padding?

fritzSF replied:
It's based on the text size defined for `.sd-avatar--inline__text`, so it makes sense here.

thecalcc commented on app-typescript/components/avatar/avatar.tsx:
```diff
@@ -65,16 +138,37 @@ export class Avatar extends React.PureComponent<IPropsAvatar> {
             </AvatarWrapper>
         );
+
+        // Render based on nameDisplay mode
```
comment isn't needed, also a bunch of other technical things, but I'll address it all by making a commit to cleanup the code

thecalcc commented on app-typescript/components/ShowPopup.tsx:
```diff
@@ -186,6 +186,7 @@ export class PopupPositioner extends React.PureComponent<IPropsPopupPositioner>
             <>
                 {ReactDOM.createPortal(
                     <div
+                        className="sd-popup-positioner"
```
we have to be careful making changes here since this show popup I think is used in a lot of places

fritzSF replied:
I know, I added it only to be able to target it when it is used for tooltips.

thecalcc reviewed (commented):
looks nice, but weren't we going to show the names to the left of the avatar icon? this is only for inline mode

fritzSF commented:
> looks nice, but weren't we going to show the names to the left of the avatar icon? this is only for inline mode

We have both options. Showing the text on the left makes sense only if it's placed to the right of a parent element (as we need for STT now). If the avatar component were on the left side of the list, it would not look good—different name lengths would push the avatar circle to the right, causing misalignment in the list.

fritzSF committed:
Remove AvatarWrapper dependency from Avatar component

fritzSF commented:
### Latest commit

Refactor Avatar component to remove AvatarWrapper dependency, improving performance and simplifying the component architecture.

### What has changed

- Removed AvatarWrapper dependency from Avatar component - Avatar now renders its own structure directly
- Converted Avatar from class component to functional component with React.memo for better performance
- Replaced nameDisplay mode 'title' with 'none' (no name display)
- Updated AvatarGroup to use Avatar with customContent for the "+N" counter instead of AvatarWrapper
- Added useMemo hooks to optimize rendering of avatar content and className

fritzSF committed:
CSS fix

thecalcc commented:
@fritzSF could you switch the branch to use something other than `develop`?

fritzSF commented:
> @fritzSF could you switch the branch to use something other than `develop`?

Done!

thecalcc commented:
closed in favour of https://github.com/superdesk/superdesk-ui-framework/pull/1001
